instance/models/utils.py

```diff
@@ -608,6 +616,99 @@ def put(self, key, value, **kwargs):
+        As of June/2019 we store configuration values using the instance's prefix as key and the value as a JSON object.
+        By using only one key we avoid the need to use transactions and to have extra logic to read/write the
+        configuration. We still need this method to make sure we increment the version number when the configuration
+        value is updated.
+
+        .. note::
+            Once all instances' configuration are updated we can retire
+            some of the code in this class.
+
+        .. note::
+            This method also works as a PATCH if only a partial payload is
+            given. To delete a key use :meth:`delete_dict_key`
+
+        :param (dict) value: value to write
+        :param (int) num_retries: Retries for self.txn_put
+        :param kwargs: consul.kv.put specific options.
+        """
+        get_data = None
+        try:
+            data_index, get_data = self._client.kv.get(self.prefix, **kwargs)
+        except consul.base.NotFound:
+            pass
+        try:
+            # This is only needed for backwards compatibility.
+            data_index, txn_data = self._client.kv.get(self.prefix, recurse=True, **kwargs)
+        except consul.base.NotFound:
+            pass
+
+        if not get_data and txn_data:
+            return self.txn_put(value, num_retries)
+
+        if get_data:
+            stored = json.loads(get_data['Value'].decode('utf-8'))
+            updates = {
+                k: value[k] for k in value
+                if (k in stored and
+                    value[k] != stored[k]) or
+                k not in stored
+            }
+            payload = {
+                k: stored[k] for k in stored
+                if k not in value and
+                k != 'version'
+            }
+            updated = bool(updates)
+            payload['version'] = stored['version']
+            payload.update(value)
+        else:
+            payload = value
+            updated = True
+
+        if updated:
+            payload['version'] = payload.get('version', 0) + 1
+
+            return payload['version'], self._client.kv.put(self.prefix, json.dumps(payload).encode('utf-8'))
+
+        return payload['version'], False
+
+    def delete_dict_key(self, key, **kwargs):
+        """
+        Delete a given key from a stored config dictionary.
+
+        :param (str) key: Key to delete
+        :param (dict) kwargs: Extra parameters for consul.kv.put
+        """
+        data_index, get_data = self._client.kv.get(self.prefix)
+        stored = json.loads(get_data['Value'].decode('utf-8'))
+        stored.pop(key)
+        stored['version'] = stored['version'] + 1
+        return self._client.kv.put(self.prefix, json.dumps(stored).encode('utf-8'))
+
+    def remove_dict(self):
+        """
+        Remove a config dict using this instance's prefix.
+        """
+        self._client.kv.delete(self.prefix)
+
+    def consolidate_values_to_dict(self):
```

Comment:
I expected this to be used to move from the old style to the new style, but found no references to this. Is that expected?

Reply:
@viadanna Yes, since the code is backwards compatible then we don't really need to run this.
This is just here as a utility function and meant to be used manually if needed.
Or depends on how we plan to move every instance to the new style.